blobfuse2-perf.yaml

```diff
@@ -0,0 +1,140 @@
+      - script: |
+          sudo rm -rf $(ROOT_DIR)
+          sudo mkdir -p $(ROOT_DIR)
+          sudo chown -R `whoami` $(ROOT_DIR)
+          chmod 777 $(ROOT_DIR)
+          mkdir -p $(ROOT_DIR)/go/src
+        displayName: 'Create Directory Structure'
+  
+      # Checkout the code  
+      - script: |
+          git clone https://github.com/Azure/azure-storage-fuse
+        displayName: 'Checkout Code'
+        workingDirectory: $(ROOT_DIR)/go/src
+
+      # Pull the branch
+      - script: |
+          git checkout `echo $(Build.SourceBranch) | cut -d "/" -f 1,2 --complement`
+        displayName: 'Checkout Branch'
+        workingDirectory: $(WORK_DIR)
+
+      # -------------------------------------------------------
+      # Pull and build the code
+      - template: 'azure-pipeline-templates/build.yml'
+        parameters:
+          working_directory: $(WORK_DIR)
+          root_dir: $(ROOT_DIR)
+          mount_dir: $(MOUNT_DIR)
+          temp_dir: $(TEMP_DIR)
+          gopath: $(GOPATH)
+          container: cont1
+          skip_ut: true
+
+      - script: |
+          cd $(WORK_DIR)
+          $(WORK_DIR)/blobfuse2 gen-test-config --config-file=azure_key.yaml --container-name=cont1 --temp-path=$(TEMP_DIR) --output-file=$(BLOBFUSE2_CFG)
+          $(WORK_DIR)/blobfuse2 mount $(MOUNT_DIR) --config-file=$(BLOBFUSE2_CFG) --default-working-dir=$(WORK_DIR)
+          python3 $(WORK_DIR)/test/perf_test/resnet50_classify.py --dataset='$(MOUNT_DIR)/data' --job='main' --log=$(ROOT_DIR)/blobfuse2-perf.log
+        displayName: "Blobfuse2 ResNet50 on Main"
+        env:
+          NIGHTLY_STO_ACC_NAME: $(PERF_WEEKLY_STO_BLOB_ACC_NAME)
+          NIGHTLY_STO_ACC_KEY: $(PERF_WEEKLY_STO_BLOB_ACC_KEY)
+          ACCOUNT_TYPE: 'block'
+          ACCOUNT_ENDPOINT: 'https://$(PERF_WEEKLY_STO_BLOB_ACC_NAME).blob.core.windows.net'
+        continueOnError: false
+
+      - script: |
+          sudo fusermount -u ${MOUNT_DIR}
+          sudo kill -9 `pidof blobfuse2` || true
+        displayName: "Unmount Blobfuse2 Main Branch Run"
+
+      - script: |
+          cd $(ROOT_DIR)
+          wget https://github.com/Azure/azure-storage-fuse/releases/download/blobfuse2-2.0.0-preview.1/blobfuse2-2.0.0-preview.1-ubuntu-20.04-x86-64.deb
+          sudo dpkg -i $(ROOT_DIR)/blobfuse2-2.0.0-preview.1-ubuntu-20.04-x86-64.deb
+          sudo apt-get install -f
+          sudo apt-get install fuse3
+          blobfuse2 mount $(MOUNT_DIR) --config-file=$(BLOBFUSE2_CFG) --default-working-dir=$(WORK_DIR)
+          cd $(WORK_DIR)
+          python3 $(WORK_DIR)/test/perf_test/resnet50_classify.py --dataset='$(MOUNT_DIR)/data' --job='binary' --log=$(ROOT_DIR)/blobfuse2-perf.log
+        displayName: "Blobfuse2 ResNet50 Binary"
+        continueOnError: false
+
+      - script: |
+          cd $(WORK_DIR)
+          python3 $(WORK_DIR)/test/perf_test/generate_perf_report.py --metrics=images/second --log=$(ROOT_DIR)/blobfuse2-perf.log
```

Comment:
Report generated by this step shall clearly tell with test1 result X, with test2 result Y and deviation Z. Also publish the same report in artifact so we dont need to go and check the results from pipeline stages.
Also we can fail this pipeline if the new numbers are falling beyond 2-3% compared to previous numbers (or test1 numbers)

Reply:
Done